frame/contracts/src/benchmarking.rs

```diff
@@ -194,78 +511,1323 @@ benchmarks! {
+		let r in 0 .. API_BENCHMARK_BATCHES;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_sha2_256", r * API_BENCHMARK_BATCH_SIZE, 0,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// `n`: Input to hash in kilobytes
+	seal_hash_sha2_256_per_kb {
+		let n in 0 .. max_pages::<T>() * 64;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_sha2_256", API_BENCHMARK_BATCH_SIZE, n * 1024,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// Only the overhead of calling the function itself with minimal arguments.
+	seal_hash_keccak_256 {
+		let r in 0 .. API_BENCHMARK_BATCHES;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_keccak_256", r * API_BENCHMARK_BATCH_SIZE, 0,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// `n`: Input to hash in kilobytes
+	seal_hash_keccak_256_per_kb {
+		let n in 0 .. max_pages::<T>() * 64;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_keccak_256", API_BENCHMARK_BATCH_SIZE, n * 1024,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// Only the overhead of calling the function itself with minimal arguments.
+	seal_hash_blake2_256 {
+		let r in 0 .. API_BENCHMARK_BATCHES;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_blake2_256", r * API_BENCHMARK_BATCH_SIZE, 0,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// `n`: Input to hash in kilobytes
+	seal_hash_blake2_256_per_kb {
+		let n in 0 .. max_pages::<T>() * 64;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_blake2_256", API_BENCHMARK_BATCH_SIZE, n * 1024,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// Only the overhead of calling the function itself with minimal arguments.
+	seal_hash_blake2_128 {
+		let r in 0 .. API_BENCHMARK_BATCHES;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_blake2_128", r * API_BENCHMARK_BATCH_SIZE, 0,
+		), vec![], Endow::Max)?;
+		let origin = RawOrigin::Signed(instance.caller.clone());
+	}: call(origin, instance.addr, 0.into(), Weight::max_value(), vec![])
+
+	// `n`: Input to hash in kilobytes
+	seal_hash_blake2_128_per_kb {
+		let n in 0 .. max_pages::<T>() * 64;
+		let instance = instantiate_contract::<T>(hasher_code(
+			"seal_hash_blake2_128", API_BENCHMARK_BATCH_SIZE, n * 1024,
```

Comment:
Why is it 1024?

Reply:
Cause `n` is defined in kilobytes (64k in one page).